onnx/optimizer/passes/fuse_add_bias_into_conv.h

```diff
@@ -15,140 +15,143 @@
 
 #include <numeric>
 
-#include "onnx/optimizer/passes/optimize_pass.h"
+#include "onnx/common/assertions.h"
+#include "onnx/optimizer/pass.h"
 
-namespace ONNX_NAMESPACE { namespace optimization {
+namespace ONNX_NAMESPACE {
+namespace optimization {
 
-struct FuseAddBiasIntoConv final : public OptimizePass {
+struct FuseAddBiasIntoConv final : public PredicateBasedPass {
```

Comment:
Although this one can be considered as a predicatebase pass, however, it's very complicated, I would suggest to make it pass to make sure, the iterator will be valid...

Reply:
I think we have one to one parity with the old implementation when using NodeDestroyType. It would make the code pretty messy especially if we want to use CountBasedAnalysis. Are you sure we should rewrite it using FullGraphBasedPass?

Reply:
I am fine to keep this as PredicateBasedPass, just want to make sure the logic is correct, since we did pretty complicated manipulations on the graph. I will double check the implementation on this again.